### PR TITLE
Apply Traits to Communication

### PR DESCRIPTION
**Pull request by kasobol-msft**
This works spans from https://github.com/Azure/azure-sdk-for-java/pull/26562  where interfaces to capture commonly occurring APIs at client builders has been created.

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientBuilderTest.java [hunk 1]
```diff
@@ -302,6 +302,22 @@ public void addPolicyNullThrows() {
         });
     }
 
+    @Test
+    public void bothRetryOptionsAndRetryPolicySetSync() {
+        assertThrows(IllegalStateException.class, () -> setupBuilderWithHttpClientWithCredential(this.clientBuilder)
```
Comment (whisper6284) [inline sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersClientBuilderTest.java:307]: I don't understand these two tests. Why is IllegalStateException thrown?

**Review by Shamkh — APPROVED** [review]

**Review by petrsvihlik — CHANGES_REQUESTED** [review]
Identity 
- ✅ code changes lgtm 
- ❌ missing a changelog entry

File: sdk/communication/azure-communication-callingserver/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Release History
 
 ## 1.0.0-beta.5 (Unreleased)
-
+- Added interfaces from `com.azure.core.client.traits` to `CallingServerClientBuilder`
+-
```
Comment (srnagar) [inline sdk/communication/azure-communication-callingserver/CHANGELOG.md:5]: Remove this line (or call out the new retryOptions API on the builder)

**Review by srnagar — APPROVED** [review]

File: sdk/communication/azure-communication-identity/src/main/java/com/azure/communication/identity/CommunicationIdentityClientBuilder.java [hunk 3]
```diff
@@ -161,31 +223,66 @@ public CommunicationIdentityClientBuilder clientOptions(ClientOptions clientOpti
...
+     * trait that are also ignored if an {@link HttpPipeline} is specified, so please be sure to refer to the
+     * documentation of types that implement this trait to understand the full set of implications.</p>
+     * <p>
+     * Setting this is mutually exclusive with using {@link #retryPolicy(RetryPolicy)}.
+     *
+     * @param retryOptions The {@link RetryOptions} to use for all the requests made through the client.
+     * @return The updated {@link CommunicationIdentityClientBuilder} object.
+     */
+    @Override
+    public CommunicationIdentityClientBuilder retryOptions(RetryOptions retryOptions) {
```
Comment (AikoBB) [inline sdk/communication/azure-communication-identity/src/main/java/com/azure/communication/identity/CommunicationIdentityClientBuilder.java:284]: could you pls mention the addition of this setter in the CHANGELOG

**Review by AikoBB — APPROVED** [review]

**Review by petrsvihlik — APPROVED** [review]

**Pull request merged**